scene/resources/resource_format_text.cpp

```diff
@@ -1507,9 +1505,7 @@ String ResourceFormatLoaderText::get_resource_script_class(const String &p_path)
 }
 
 ResourceUID::ID ResourceFormatLoaderText::get_resource_uid(const String &p_path) const {
-	String ext = p_path.get_extension().to_lower();
-
-	if (ext != "tscn" && ext != "tres") {
+	if (p_path.has_extension("tscn") && !p_path.has_extension("tres")) {
```

Comment:
This is arguably a reduction in performance as it turns one `get_extension().to_lower()` into two, I'd not replace this case if the goal is performance (I think both have equivalent readability)

Also typo:
```suggestion
	if (!p_path.has_extension("tscn") && !p_path.has_extension("tres")) {
```

Reply:
`has_extension()` does not perform `get_extension().to_lower()` directly, but yeah, I didn't check which one is faster (might be the new one).